src/main/java/com/minecolonies/coremod/entity/ai/mobs/EntityAIAttackRangedBowBarbarian.java

```diff
@@ -0,0 +1,189 @@
+package com.minecolonies.coremod.entity.ai.mobs;
+
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.ai.EntityAIBase;
+import net.minecraft.init.Items;
+import net.minecraft.item.ItemBow;
+import net.minecraft.util.EnumHand;
+
+/**
+ * Class implementing the AI for the Archer Barbarians ranged attack.
+ */
+public class EntityAIAttackRangedBowBarbarian extends EntityAIBase
+{
+    private final AbstractArcherBarbarian entity;
+    private final double                  moveSpeedAmp;
+    private       int                     attackCooldown;
+    private final float                   maxAttackDistance;
+    private int attackTime = -1;
+    private int     seeTime;
+    private boolean strafingClockwise;
+    private boolean strafingBackwards;
+    private int strafingTime = -1;
+
+    public EntityAIAttackRangedBowBarbarian(final AbstractArcherBarbarian archer,final double speedAmplifier,final int delay,final float maxDistance)
+    {
+        super();
+        this.entity = archer;
+        this.moveSpeedAmp = speedAmplifier;
+        this.attackCooldown = delay;
+        this.maxAttackDistance = maxDistance * maxDistance;
+        this.setMutexBits(3);
+    }
+
+    public void setAttackCooldown(final int p_189428_1_)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00117)